### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by Holzhaus**
Follow up to #1. No code changes, just cleaning up the repo:
- Improve directory structure
- Add README
- Rename lgpl.txt to LICENSE
- Removes binaries and junk files
- Remove doxygen HTML docs. Instead the HTML are now built on GitHub Actions and deployed to GitHub pages.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,7 +1,7 @@
-		   GNU LESSER GENERAL PUBLIC LICENSE
+                   GNU LESSER GENERAL PUBLIC LICENSE
                        Version 3, 29 June 2007
 
- Copyright (C) 2007 Free Software Foundation, Inc. <http://fsf.org/>
+ Copyright (C) 2007 Free Software Foundation, Inc. <https://fsf.org/>
```
Comment (Be-ing) [inline LICENSE:4]: lol that's quite the attention to detail

File: README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# HSS1394
+
+*High-speed MIDI-over-Firewire device access library for Windows and Mac OSX.*
```
Comment (Be-ing) [inline README.md]: macOS

**Comment by Be-ing** [comment]
How about moving src/osx/1394API_OSX.cpp to src/macos/1394API_macOS.cpp

File: LICENSE (hunk 1 above)
Comment (Holzhaus) [reply LICENSE:4]: Haha no I noticed mixed tabs and spaces and just redownloaded the license file from gnu.org :D

File: CMakeLists.txt [hunk 3]
```diff
@@ -18,7 +18,7 @@ target_include_directories(hss1394 PRIVATE
 )
 
 if(APPLE)
-  target_sources(hss1394 PRIVATE src/osx/1394API_OSX.cpp)
+  target_sources(hss1394 PRIVATE src/macos/1394API_OSX.cpp)
```
Comment (Be-ing) [inline CMakeLists.txt]: and the filename: 1394API_MAC.cpp or 1394_MACOS.cpp or 1394_macOS.cpp. Take your pick

**Comment by Be-ing** [comment]
Can you deploy builds as GH Actions artifacts?

**Comment by Holzhaus** [comment]
> Can you deploy builds as GH Actions artifacts?

I have no idea how to package binaries on macOS/Windows. Usually I'd just use a tar.gz or zip file, but I think the cpack TGZ generator is for source files, not binaries. I'll have to google that. Anyway, that can be part of another PR and should not block this one.

**Comment by Be-ing** [comment]
Now I remember the other reason for using macOS 10.15 on GH Actions for Mixxx. The macOS 11.0 jobs wait in the queue a long time before executing.

**Comment by Be-ing** [comment]
For libKeyFinder I just upload the whole installation directory as the GH Actions artifact. GH Actions automatically puts it in a ZIP file. No need for anything fancier.
https://github.com/mixxxdj/libKeyFinder/blob/master/.github/workflows/build-and-test.yml#L38

**Comment by Be-ing** [comment]
Can you rebase to remove 3a32e4f?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# HSS1394
+
+*High-speed MIDI-over-Firewire device access library for Windows and macOS.*
+
+HSS1394 is a high-speed MIDI-over-Firewire protocol developed by Stanton Magnetics, Inc.
+This project contains the LGPLv3 library code that interfaces with HSS1394 devices on Windows and macOS, such as the Stanton SCS.1 controller series.
```
Comment (Be-ing) [inline README.md]: This library is maintained by the [Mixxx DJ Software](https://mixxx.org/) team. Contributions are welcome by opening pull requests and issues on [GitHub](https://github.com/mixxxdj/hss1394). If you want to discuss anything about this library with us, please get in touch on our [Zulip chat](https://mixxx.zulipchat.com/).

Comment (Be-ing) [reply README.md]: ## Building

HSS1394 uses the [CMake build system](https://cmake.org/) and follows standard CMake procedures. From the top directory of this source code repository, run:

```
cmake -DCMAKE_INSTALL_PREFIX=/path/to/install/to -S . -B build
cmake --build build
cmake --install build
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+HSS1394 uses the [CMake build system](https://cmake.org/).
+From the top directory of this source code repository, run:
+
+    $ cmake -DCMAKE_INSTALL_PREFIX=/path/to/install/to -S . -B build
+    $ cmake --build build
+    $ cmake --install build
+
+## License
+
+The HSS1394 library is licensed under the terms of the GNU Lesser General Public License (LGPL) version 3.
```
Comment (Be-ing) [inline README.md]:
```suggestion
The HSS1394 library is licensed under the terms of the GNU Lesser General Public License (LGPL) version 3 or later.

```

**Comment by Be-ing** [comment]
Besides adding the "or later", this looks good to me. I don't have write access to this repository, so I can't merge.

File: README.md (hunk 5 above)
Comment (Be-ing) [reply README.md]: fixed in #3

**Comment by Be-ing** [comment]
Should this be rebased now?

**Comment by Holzhaus** [comment]
> Should this be rebased now?

Done.

**Comment by Be-ing** [comment]
I don't have permission to update the repo settings. Can you turn on GitHub Pages?

**Comment by Holzhaus** [comment]
> I don't have permission to update the repo settings. Can you turn on GitHub Pages?

Done. https://mixxxdj.github.io/hss1394/

**Comment by Be-ing** [comment]
Can you update the repo settings to list https://mixxxdj.github.io/hss1394/ as the project website?

**Comment by Holzhaus** [comment]
> Can you update the repo settings to list https://mixxxdj.github.io/hss1394/ as the project website?

I'll do that tomorrow. I'm currently on my phone and it's neither possible to add a website in the mobile view nor to force the desktop website on my phone :(

**Pull request merged**